README.md
Backend Simplifier (tsconf)
===========================
The Backend Simplifier offers the configuration of the eight most commonly used
properties for simplifying the TYPO3 backend by a mouse click. User interface
is the extension manager. Enable page tree uids, use indvidual page tree icons,
activate the extended view and the clipboard and the admin panel for all editors.
Extend the cache thunderbolts with one for the system cache.
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
Versions 0.0.4 - 9.0.2 have been completely provided by Dirk Wildt (Die Netzmacher).   
These Versions have been downloaded here: https://extensions.typo3.org/extension/tsconf/  

The versions provided here are copies but versioned and tagged with git,
with this additional text-file `README.md` and **composer-support**.  

To install via composer use:  
```
composer require wdb/tsconf
```

To read the original `README.md`-file, open `README_original.md`  


- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
9.0.2 / stable  
Supported TYPO3 version: 6.2.0 - 9.0.99  
November 25, 2018  
Compatible with TYPO3 9.5 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
9.0.0 / stable  
Supported TYPO3 version: 6.2.0 - 9.0.99  
December 30, 2017  
**Upgrade for TYPO3 9.0.**  
 - This version has a bug: The tsconf page tree icons won't displayed. I [Dirk Wildt] will fix it soon as possible.  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
7.2.4 / stable  
Supported TYPO3 version: 6.2.0 - 8.9.99  
December 29, 2017  
**Improvement:**  
 - Move some code from ext_tables.php to TCA/Overrides.  
 - Bugfix: extension icon was missing.  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
7.2.2 / stable  
Supported TYPO3 version: 6.2.0 - 8.9.99  
October 05, 2017  
**IMPROVEMENT**  
 - Extension manager > Reduce headers: Header 1 will not longer removed. This is in respect to fluid_styled_content. 0 (Default) is wrapped with h2.  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
7.2.1 / stable  
Supported TYPO3 version: 6.2.0 - 8.9.99  
October 01, 2017  
 - BUGFIX for TYPO3 6.x  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
7.2.0 / stable  
Supported TYPO3 version: 6.2.0 - 8.9.99  
September 30, 2017  
 - UPGRADE for TYPO3 8.7.  
 - Compatible with TYPO3 6.2 and 7.6.  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
7.1.1 / stable  
Supported TYPO3 version: 6.2.0 - 7.6.99  
December 14, 2016  
 - TYPO3 7.6 Compatibility.  
 - Bugfix pagetree icons.  
 - ReST documentation.  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
7.0.0 / stable  
Supported TYPO3 version: 6.0.0 - 7.6.99  
June 28, 2016  
**COMPATIBILITY:**  
 - TYPO3 7.x is supported  
 - TYPO3 4.x isn't supported any longer.  
**Removed:**  
 - simplify page properties.  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
6.2.1 / stable  
Supported TYPO3 version: 4.5.0 - 7.0.99  
October 21, 2015  
**Bugfix:**
 - JavaScript for enhanced page tree menu was misssing  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
6.2.0 / stable  
Supported TYPO3 version: 4.5.0 - 7.0.99  
October 06, 2015  
**New feature:**  
 - pagetree with enhanced context menu. Thanks to Andreas Kießling/pluspol for this new feature.  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
6.1.0 / stable  
Supported TYPO3 version: 4.5.0 - 7.0.99  
April 01, 2015  
**New Feature:**  
 - Own Icons. Bugfix: TYPO3 4.x  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
6.0.1 / stable  
Supported TYPO3 version: 4.5.0 - 7.0.99  
January 20, 2015  
**Compatibility** from TYPO3 4.5 to 7.0  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
6.0.0 / stable  
Supported TYPO3 version: 6.0.0 - 7.0.99  
December 16, 2014  
**New feature:**  
 - default upload folder  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
1.6.0 / stable  
Supported TYPO3 version: 6.0.0 - 7.0.99  
December 09, 2014  
**Compatible** with TYPO3 7.0  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
1.5.2 / stable  
Supported TYPO3 version: 6.0.0 - 6.2.99  
November 22, 2014  
**Bugfix:**  
 - Class t3lib_extMgm is moved to \TYPO3\CMS\Core\Utility\ExtensionManagementUtility
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
1.5.1 / stable  
Supported TYPO3 version: 6.0.0 - 6.2.99  
October 29, 2014  
**Bugfix:**  
 - 1.5.x is running in TYPO3 from 6.x only.  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
1.5.0 / stable  
Supported TYPO3 version: 4.5.0 - 6.2.99  
October 27, 2014  
**tt_content:**  
 - enhance the maximum for managing images from 7 to 12. Enables image cutting.  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
1.4.4 / stable  
Supported TYPO3 version: 4.5.0 - 6.2.99  
July 07, 2014  
**NEW FEATURES:**  
 - default localisation icon and label  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
1.4.3 / stable  
Supported TYPO3 version: 4.5.0 - 6.2.99  
June 05, 2014  
**Improvement:**  
 - &Uuml;berschrift -> Ãœberschrift  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
1.4.2 / stable  
Supported TYPO3 version: 4.5.0 - 6.2.99  
June 01, 2014  
**NEW FEATURE:**  
 - options.clearCache.system is added and is enabled by default  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
1.4.0 / stable  
Supported TYPO3 version: 4.5.0 - 6.2.99  
May 07, 2014  
**NEW FEATURE:**  
 - TYPO3 6.2  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
1.3.1 / stable  
Supported TYPO3 version: 4.5.0 - 6.1.99  
January 25, 2014  
**FEATURE:**  
 - Big form fields for including TypoScript static templates  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
1.3.0 / stable  
Supported TYPO3 version: 4.5.0 - 6.1.99  
January 25, 2014  
**FEATURE:**  
 - Big form fields for including TypoScript static templates  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
1.2.0 / stable  
Supported TYPO3 version: 4.5.0 - 6.1.99  
January 13, 2014  
**Feature:**
 - TYPO3 6.1  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
1.1.1 / stable  
February 13, 2013  
**New icons:**  
 - backend_layout, jQuery and TYPO3  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
1.1.0 / stable  
February 01, 2013  
**IMPROVEMENT:**  
 - Downgrade for TYPO3 4.4  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
1.0.1 / stable  
March 20, 2012  
**IMPROVEMENT:**  
 - Usage of $ICON_TYPES is deprecated since 4.4  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
1.0.0 / stable  
February 23, 2012  
**NEW FEATURES**
 - 34 new icons  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
0.1.3 / stable  
May 05, 2011  
**IMPROVEMENTS:**  
 - 27 new icons: flags, folders and library.  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
0.1.2 / stable  
March 07, 2011  
**New feature:**  
 - Page tree: TypoScript template icon
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
0.1.1 / stable  
February 27, 2011  
**SEO:**  
 - Extension title is moved from tsConf to tsConfig  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
0.1.0 / stable  
February 27, 2011  
**NEW FEATURES:**  
 - Module web list: activate Extended view and Show clipboard by default.  
 - Module file list: activate Extended view, Display thumbnails and Show clipboard by default  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
0.0.7 / stable  
February 22, 2011  
 - **Corporate Design:** Netzmacher icon is moved to TYPO3 icon  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
0.0.5 / stable  
January 20, 2011  
**INITIAL RELEASE:** Configure your default userTSconf and pageTSconf supported by the extension manager!  
**BUGFIX:** proper manual  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
0.0.4 / stable  
January 20, 2011  
**INITIAL RELEASE:** Configure your default userTSconf and pageTSconf supported by the extension manager  


ChangeLog
9.0.2
  18-11-25
	Improvement
  * #i0025: Compatible with TYPO3 9.5

9.0.1
  2018-09-11
	Improvement
	* #i0024: labelling ext_conf_template.txt

9.0.0
  2017-12-30
	BUG: The tsconf page tree icons won't displayed. I will fix it soon as possible.
	Feature
	* #i0023.9: Upgrade for TYPO3 9.0
							* ext_emconf.php
							* ext_conf_template.txt: options > option

7.2.4
  2017-12-29
	Improvement
	* #i0022: Move some code from ext_tables.php to TCA/Overrides
	Bugfix
	* #i0021: Extension icon was missing 

7.2.3
  2017-10-20
	Improvement
	* #i0020: Icon for E-mail (from TYPO3 8.7 only)

7.2.2
  2017-10-02
	Improvement
	* #i0019: Extension manager > Reduce headers
						* Had removed 1,4 and 5. Is changed to 4 and 5 only
						* This is in respect to fluid_styled_content:
							* 0 (Default) is wrapped with h2

7.2.1
  2017-10-01
	Bugfix
	* #i0018: IconRegistry error in TYPO3 6.x

7.2.0
  2017-09-30
	Upgrade for TYPO3 8.x
	* #t1576: SpriteManager > IconRegistry

7.1.2
  16-12-29
  Bugfix
  * #i0017: 

7.1.1
  16-12-14
  Improvement
  * #i0016: ReST documentation
  * #i0015: Downward compatibility TYPO3 6.2

7.1.0
  Sorry, but NO SUPPORT TYPO3 6.x any longer. 
  16-11-01
  Improvement
  * #i0014: Bugfix pagetree icons

7.0.1
  16-06-26
  Improvement
  * #i0013: Language Overlay

7.0.0
  16-06-25
  Improvement
  * #i0012: TYPO3 7.6 Compatibility.
            TYPO3 4.x isn't supported any longer.
            Simplify page properties is removed.


6.3.1
  16-01-05
  Improvement
  * #i0011: Clean up file structure

6.3.0
  16-01-04
  Improvement
  * #i0010: Update for TYPO3 7.6
            * t3lib_extMgm is moved to \TYPO3\CMS\Core\Utility\ExtensionManagementUtility
            * ClickMenuAction and PagetreeNode is disabled because of class not found error

6.2.1
  15-10-21
  Bugfix
  * #i0009: JavaScript code was missing for enhanced context menu of the pagetree

6.2.0
  15-10-06
  Feature
  * #70445: Enhances the context menu of the pagetree
  Improvement
  * #i0009: Default value of page_tceform_pages is moved from 1 to 0

6.1.0
  15-03-31
  Feature
  * #66176: Individual Icons
  Bugfix
  * #66211: No items in TYPO3 4.x

6.0.1
  15-01-20
  Feature
  * #i0008: Compatibility for TYPO3 4.x

6.0.0
  14-12-16
  Feature
  * #i0007: Default upload folder

1.6.0
  14-10-27
  Feature
  * #i0006: TYPO3 7.0
            * TYPO3\CMS\Backend\Sprite\SpriteManager

1.5.2
  14-11-22
  Bugfix
  * #i0006: Class \TYPO3\CMS\Core\Utility\ExtensionManagementUtility is moved to \TYPO3\CMS\Core\Utility\ExtensionManagementUtility

1.5.0
  14-10-27
  New Feature
  * #62476: Enhance the maximum of the image property columns (imgcols) from 7 to 12
	* #62475: Move the image properties height and width from integer to string to enable values like 100c.

1.4.4
  14-07-07
  New Feature
  * #i0005: mod.SHARED.defaultLanguageFlag and mod.SHARED.defaultLanguageLabel

1.4.3
  14-06-04
  Improvement
  * #i0004: &Uuml;berschrift -> Ueberschrift

1.4.2
  14-05-31
  New Feature
  * #i0003: options.clearCache.system

1.4.1
  14-05-07
  Improvement
  * #i0002: Author ext_emconf.php

1.4.0
  14-05-07
  Feature
  * #58585: Fit it into TYPO3 6.2

1.3.0
  14-01-25
  Feature
  * 55323: Big form fields for including TypoScript static templates

1.2.0
  14-01-13
  Feature
  * #54966: TYPO3 6.1

1.1.1
  13-02-12
  Features
  * #i0001: New icons: backend_layout, jQuery and TYPO3
            Effected files:
            * ext_tables.php
            * ext_icons
              * backend_layout.gif
              * jQuery.gif
              * typo3.png

1.1.0
  13-02-01
  Improvement
  * #45079: Downgrade for TYPO3 < 4.4
            Effected file:
            * ext_tables.php

1.0.1
  12-03-20
  Improvement
  * #34858: Usage of $ICON_TYPES is deprecated since 4.4.

1.0.0
  New feature
  12-02-23
  * X new icons:  application-exit, applications-development, applications-internet, application-rss+xml,
                  call-start, call-stop, camera-photo, chronometer, code-class, code-context, configure,
                  dialog-cancel, dialog-close, dialog-ok-applay, dialog-password, edit-bomb, emblem-favorite,
                  emblem-important, games-highscores, help-hint, home-mdk, media-optical-recordable,
                  office-calendar, preferences-system, roll, security-high, security-low, security-medium,
                  speaker, status-unknown, tool-animator, user-trash, view-conversation-balloon, view-filter

0.1.3
  11-05-05
  Improvements
  * 27 new icons: flags, folders and library

0.1.2
  11-03-07
  New feature
  * Page tree: TypoScript template icon

0.1.1
  11-02-27
  Improvement
  * Extension title is moved from tsConf to tsConfig (search engine optimization)

0.1.0
  11-02-26
  New Features
  * Module web list: activate Extended view and Show clipboard by default
  * Module file list: activate Extended view, Display thumbnails and Show clipboard by default

0.0.7
  11-02-22
  Corporate Design
  * Netzmacher icon is moved to TYPO3 icon

0.0.6
  11-01-26
  Improvement
  * Added to user TSconf
    * options.clearCache.pages

0.0.5
  11-01-20
  Bugfix
  * Proper Manual

0.0.4
  11-01-19
  Improvement
  * extkey moved from my_tsconf to tsconf
  * page and user TSconf can be configured per item by the extension manager
  Publishing

0.0.3
  11-01-19
  Improvement
  * extkey moved from my_usertsconf to my_tsconf
  * page TSconf moved from ext_tables.php to pageTSconfig.txt
  * user TSconf moved from ext_tables.php to userTSconfig.txt
  * page and user TSconf can be disabled by the extension manager
  * Added
    * mod.web_layout.menu.function.0 = 0 (Quick Edit)
    * mod.web_layout.menu.function.3 = 0 (Page Information)
    * TCEFORM.pages.alias.disabled = 1
    * TCEFORM.pages.content_from_pid.disabled = 1
    * TCEFORM.tt_content.header_layout: new labels
    * TCEFORM.tt_content.header_layout.removeItems = 1, 4, 5

0.0.2
  11-01-19
  Improvement
  * Added
    * options.clearCache.pages
    * options.clearCache.all

0.0.1
  11-01-19

ext_emconf.php
<?php

$EM_CONF[ $_EXTKEY ] = array(
  'title' => 'Backend Simplifier',
  'description' => 'The Backend Simplifier offers the configuration of the eight most commonly used porperties for simplifying the TYPO3 backend by a mouse cklick. User interface is the extension manager. Enable page tree uids, use indvidual page tree icons, activate the extended view and the clipboard and the admin panel for all editors. Extend the cache thunderbolts with one for the system cache.',
  'category' => 'be',
  'shy' => 0,
  'version' => '9.0.2',
  'dependencies' => '',
  'conflicts' => '',
  'priority' => '',
  'loadOrder' => '',
  'module' => '',
  'state' => 'stable',
  'uploadfolder' => 0,
  'createDirs' => '',
  'modify_tables' => '',
  'clearcacheonload' => 0,
  'lockType' => '',
  'author' => 'Dirk Wildt (Die Netzmacher)',
  'author_email' => 'http://wildt.at.die-netzmacher.de',
  'author_company' => '',
  'CGLcompliance' => '',
  'CGLcompliance_note' => '',
  'constraints' => array(
    'depends' => array(
      'typo3' => '6.2.0-9.9.99',
    ),
    'conflicts' => array(
    ),
    'suggests' => array(
    ),
  ),
  '_md5_values_when_last_written' => 'a:83:{s:9:"ChangeLog";s:4:"2247";s:21:"ext_conf_template.txt";s:4:"8a78";s:12:"ext_icon.gif";s:4:"25cc";s:17:"ext_localconf.php";s:4:"aaf7";s:14:"ext_tables.php";s:4:"2b08";s:22:"locallang_tsConfig.php";s:4:"6ec0";s:30:"pageTSconfig_tceform_pages.txt";s:4:"1dfc";s:34:"pageTSconfig_tceform_ttcontent.txt";s:4:"d1b0";s:41:"pageTSconfig_weblayout_menu_functions.txt";s:4:"877c";s:38:"pageTSconfig_weblist_enablebuttons.txt";s:4:"b5f8";s:27:"userTSconfig_adminPanel.txt";s:4:"4be4";s:33:"userTSconfig_clearCacheSystem.txt";s:4:"bc69";s:39:"userTSconfig_filelist_enablebuttons.txt";s:4:"5a62";s:30:"userTSconfig_options_cache.txt";s:4:"a2b3";s:36:"userTSconfig_options_pagetreeuid.txt";s:4:"a032";s:14:"doc/manual.pdf";s:4:"1b68";s:14:"doc/manual.sxw";s:4:"fad8";s:30:"ext_icons/application-exit.png";s:4:"f968";s:33:"ext_icons/application-rss+xml.png";s:4:"faf3";s:38:"ext_icons/applications-development.png";s:4:"345f";s:35:"ext_icons/applications-internet.png";s:4:"ce2b";s:28:"ext_icons/backend_layout.gif";s:4:"8275";s:24:"ext_icons/call-start.png";s:4:"ce14";s:23:"ext_icons/call-stop.png";s:4:"7d85";s:26:"ext_icons/camera-photo.png";s:4:"b0e4";s:25:"ext_icons/chronometer.png";s:4:"0183";s:24:"ext_icons/code-class.png";s:4:"cf37";s:26:"ext_icons/code-context.png";s:4:"a069";s:23:"ext_icons/configure.png";s:4:"0180";s:27:"ext_icons/dialog-cancel.png";s:4:"99c7";s:26:"ext_icons/dialog-close.png";s:4:"3581";s:29:"ext_icons/dialog-ok-apply.png";s:4:"d14c";s:29:"ext_icons/dialog-password.png";s:4:"c74a";s:23:"ext_icons/edit-bomb.png";s:4:"56cd";s:29:"ext_icons/emblem-favorite.png";s:4:"8e9e";s:30:"ext_icons/emblem-important.png";s:4:"97f4";s:24:"ext_icons/flag-black.png";s:4:"a049";s:23:"ext_icons/flag-blue.png";s:4:"4434";s:24:"ext_icons/flag-green.png";s:4:"6420";s:22:"ext_icons/flag-red.png";s:4:"dbf9";s:25:"ext_icons/flag-yellow.png";s:4:"c21e";s:25:"ext_icons/folder-blue.png";s:4:"7952";s:29:"ext_icons/folder-bookmark.png";s:4:"f035";s:26:"ext_icons/folder-brown.png";s:4:"6066";s:25:"ext_icons/folder-cyan.png";s:4:"df95";s:32:"ext_icons/folder-development.png";s:4:"5396";s:30:"ext_icons/folder-documents.png";s:4:"8a63";s:30:"ext_icons/folder-downloads.png";s:4:"88f4";s:30:"ext_icons/folder-favorites.png";s:4:"b2df";s:26:"ext_icons/folder-green.png";s:4:"f1bd";s:25:"ext_icons/folder-grey.png";s:4:"6c24";s:30:"ext_icons/folder-important.png";s:4:"f7ab";s:27:"ext_icons/folder-locked.png";s:4:"debd";s:27:"ext_icons/folder-orange.png";s:4:"0a98";s:24:"ext_icons/folder-red.png";s:4:"e72a";s:27:"ext_icons/folder-remote.png";s:4:"0765";s:26:"ext_icons/folder-sound.png";s:4:"642b";s:24:"ext_icons/folder-tar.png";s:4:"bb73";s:24:"ext_icons/folder-txt.png";s:4:"9f17";s:26:"ext_icons/folder-video.png";s:4:"b7c5";s:27:"ext_icons/folder-violet.png";s:4:"768c";s:27:"ext_icons/folder-yellow.png";s:4:"a96c";s:20:"ext_icons/folder.png";s:4:"b396";s:30:"ext_icons/games-highscores.png";s:4:"87d6";s:23:"ext_icons/help-hint.png";s:4:"6525";s:22:"ext_icons/home-mdk.png";s:4:"3a17";s:20:"ext_icons/jQuery.gif";s:4:"72a2";s:21:"ext_icons/library.png";s:4:"1606";s:38:"ext_icons/media-optical-recordable.png";s:4:"d57d";s:29:"ext_icons/office-calendar.png";s:4:"3820";s:32:"ext_icons/preferences-system.png";s:4:"73a5";s:18:"ext_icons/roll.png";s:4:"42e8";s:27:"ext_icons/security-high.png";s:4:"2459";s:26:"ext_icons/security-low.png";s:4:"fe8b";s:29:"ext_icons/security-medium.png";s:4:"5fb9";s:21:"ext_icons/speaker.png";s:4:"cdb5";s:28:"ext_icons/status_unknown.png";s:4:"b439";s:22:"ext_icons/template.gif";s:4:"2f14";s:27:"ext_icons/tool-animator.png";s:4:"d90e";s:19:"ext_icons/typo3.png";s:4:"d11c";s:24:"ext_icons/user-trash.png";s:4:"52b4";s:39:"ext_icons/view-conversation-balloon.png";s:4:"6b6c";s:25:"ext_icons/view-filter.png";s:4:"4d5a";}',
  'suggests' => array(
  ),
);